### PR TITLE
Pc 1313

### PR DESCRIPTION
**Pull request by eyalvo-matrix**
Only a draft

### CODE REVIEW

File: modules/GenericTools/src/GenericTools/Model/baseTable.php [hunk 1]
```diff
@@ -230,7 +230,9 @@ public function buildGenericSelect(array $FilterData = null,$order = null,$speci
 
         $where = new Where();
         foreach($FilterData as $field=>$value){
-            $last = self::$AND;
+
+
+            $last = (is_null($value[0]['sqlOp'])) ? self::$AND : $value[0]['sqlOp'];
```
Comment (matrix-amiel) [inline modules/GenericTools/src/GenericTools/Model/baseTable.php:235]: need to test that  this change works well with requests with additional parameters

Comment (eyalvo-matrix) [reply modules/GenericTools/src/GenericTools/Model/baseTable.php:235]: LIKE ? AND is default  and there is only OR .. What other param can be ?

File: modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Builders/EncounterBuilder.php [hunk 2]
```diff
@@ -131,12 +131,13 @@ class	        token	        Classification of patient encounter	Encounter.class
         $this->createMapping(["FHIREncounter",'status',1,'EncounterStatus',"","form_encounter","form_encounter.status","EncounterStatus (Required)",self::EXACT,""]);
         $this->createMapping(["FHIREncounter",'service-type',1,'EncounterStatus',"","form_encounter","lr.option_id","EncounterStatus (Required)",self::EXACT,""]);
         $this->createMapping(["FHIREncounter",'service-provider',1,'EncounterStatus',"","form_encounter","facility_id","EncounterStatus (Required)",self::EXACT,""]);
-
         $this->createMapping(["FHIREncounter",'arrival-way',1,'form_encounter',"","form_encounter","arrival_way"," ",self::PART,""]);
         $this->createMapping(["FHIREncounter",'reason-codes-details',1,'form_encounter',"","form_encounter","reason_codes_details"," ",self::PART,""]);
 
+        $this->createMapping(["FHIREncounter",'all-statuses',1,'form_encounter',"","form_encounter","all_statuses"," ",self::EXACT,""]);
```
[the remark this answers is not in the document]
Comment (eyalvo-matrix) [reply modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Builders/EncounterBuilder.php:137]: sure , I had a feeling you will change that :)

File: modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Builders/EncounterBuilder.php [hunk 3]
```diff
@@ -131,12 +131,13 @@ class	        token	        Classification of patient encounter	Encounter.class
...
         $this->createMapping(["FHIREncounter",'reason-codes-details',1,'form_encounter',"","form_encounter","reason_codes_details"," ",self::PART,""]);
 
+        $this->createMapping(["FHIREncounter",'status-extended',1,'form_encounter',"","form_encounter","all_statuses"," ",self::EXACT,""]);
+
         /////////////////////////////////////////////SORT BY id,date,appointment,patient,status    from mysql table form_encounter column id
-        $this->createMapping(["FHIREncounter",'_sort',1,'token',"id,date,appointment,patient,status,priority,service-type","form_encounter","form_encounter.id,form_encounter.date,form_encounter.eid,form_encounter.pid,form_encounter.status,form_encounter.priority,service_type_seq",'A code for the type of Encounter',self::EXACT,""]);
+        $this->createMapping(["FHIREncounter",'_sort',1,'token',"id,date,appointment,patient,status,priority,service-type,status_update_date","form_encounter","form_encounter.id,form_encounter.date,form_encounter.eid,form_encounter.pid,form_encounter.status,form_encounter.priority,service_type_seq,form_encounter.status_update_date",'A code for the type of Encounter',self::EXACT,""]);
```
Comment (matrix-amiel) [inline modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Builders/EncounterBuilder.php:140]: change the search parameter to status-update-date (not status_update_date) like all the search parameters

Comment (matrix-amiel) [inline modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Builders/EncounterBuilder.php:140]: Add new search parameter for status-update-date (with all the modifiers of date), we need it for current development

Comment (eyalvo-matrix) [reply modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Builders/EncounterBuilder.php:140]: ok but note that
all of this config needs to go away ....
This is garbage code that no one will understand in the future

Comment (eyalvo-matrix) [reply modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Builders/EncounterBuilder.php:141]: ok

**Pull request merged**